docs/3.0rc/develop/results.mdx

```diff
@@ -1,544 +1,174 @@
+- `cache_policy`: a [cache policy](task-caching#cache-policies) specifying the behavior of the task's cache.
+- `cache_key_fn`: [a function](task-caching#cache-key-functions) that configures a custom cache policy.
 
-@task
-def my_task():
-    raise ValueError()
+Similarly, setting `persist_result=True`, `result_storage` or `result_serializer` on a flow will enable
+persistence for that flow.
 
-@flow
-def my_flow():
-    try:
-        my_task()
-    except ValueError:
-        print("Oh no! The task failed.")
+<Note>
+**Enabling persistence on a flow enables persistence by default for tasks**
 
-    return True
+Enabling result persistence on a flow through any of the above keywords will also enable it for all
+tasks called within that flow by default. 
 
-my_flow()
-```
+Any settings _explicitly_ set on a task will not be overriden by the flow defaults.
 
-If you prefer to check for a failed task without using `try/except`, you may ask Prefect to return the state:
+</Note>
 
-```python
-from prefect import flow, task
+## Configuring how results are persisted
 
-@task
-def my_task():
-    raise ValueError()
+### Default persistence configuration
 
-@flow
-def my_flow():
-    state = my_task(return_state=True)
+Once result persistence is enabled, Prefect relies on a few default settings that allow you to immediately begin
+experimenting with result persistence. 
+All of these defaults are configurable and can also be explicitly set on flows and tasks through the
+[keyword arguments listed above](results#persisting-results).
 
-    if state.is_failed():
-        print("Oh no! The task failed. Falling back to '1'.")
-        result = 1
-    else:
-        result = state.result()
+By default, results are persisted locally to `~/.prefect/storage/`. The filename is computed based on the task's
+cache policy, which is typically a hash of various pieces of data and metadata. For flows, the filename is a random UUID.
+The return values are serialized by default using `cloudpickle` which can handle most (but not all!) Python objects
+and serializes them in machine-readable form.
 
-    return result + 1
+All three of these steps can be configured.
 
-result = my_flow()
-assert result == 2
-```
+### Result storage
 
-If you retrieve the result from a failed state, the exception is raised. 
-For this reason, it's often best to check if the state is failed first.
+You can configure the system of record for your results through the `result_storage` keyword argument.
+This keyword accepts any [filesystem block](blocks) or a block document slug. 
+Note that if you want your tasks to share a common cache, your result storage should be accessible by 
+the infrastructure in which those tasks run. 
+For example, a common distributed filesystem for result storage is AWS S3.
 
 ```python
 from prefect import flow, task
+from prefect.filesystems import S3
```

Comment:
I thought this would be depricated in favor of [S3Bucket](https://docs.prefect.io/latest/integrations/prefect-aws/s3/#prefect_aws.s3.S3Bucket) from the aws integration library?

Reply:
yes they're gone from 3.x as of https://github.com/PrefectHQ/prefect/pull/13410

Reply:
ahhh yes, I'll update to import from our aws package